core/src/main/scala/org/apache/spark/deploy/history/HistoryServerArguments.scala

```diff
@@ -34,26 +34,25 @@ private[history] class HistoryServerArguments(conf: SparkConf, args: Array[Strin
 
   @tailrec
   private def parse(args: List[String]): Unit = {
-    if (args.length == 1) {
-      setLogDirectory(args.head)
-    } else {
-      args match {
-        case ("--dir" | "-d") :: value :: tail =>
-          setLogDirectory(value)
-          parse(tail)
-
-        case ("--help" | "-h") :: tail =>
-          printUsageAndExit(0)
-
-        case ("--properties-file") :: value :: tail =>
-          propertiesFile = value
-          parse(tail)
-
-        case Nil =>
-
-        case _ =>
-          printUsageAndExit(1)
-      }
+    args match {
+      case ("--dir" | "-d") :: value :: tail =>
+        setLogDirectory(value)
+        parse(tail)
+
+      case ("--help" | "-h") :: tail =>
+        printUsageAndExit(0)
+
+      case ("--properties-file") :: value :: tail =>
+        propertiesFile = value
+        parse(tail)
+
+      case dir :: Nil =>
```

Comment:
IIUC this is not related to the PR description?

Reply:
No, it is related. For the last single argument, it is treated as the location of directory.
See the deleted code
```
if (args.length == 1) {
  setLogDirectory(args.head)
}
```
I prefer to keep the behavior.

Reply:
I'm not against the change, but we shall mention it in the PR desc.

Reply:
sounds good.

Reply:
I see. I have updated the description.